koach/server/koachServer.go

```diff
@@ -218,6 +309,81 @@ func (ks *KoachServer) Get(ctx context.Context, request *proto.GetRequest) (*pro
 	return &response, nil
 }
 
+func (ks *KoachServer) addAlertChans(id string, alertChan chan *model.Alert) {
+	ks.AlertLock.Lock()
+	defer ks.AlertLock.Unlock()
+
+	ks.AlertChans[id] = alertChan
+
+	kg.Printf("Added a new client (%s) for ListenAlert", id)
+}
+
+func (ks *KoachServer) removeAlertChans(id string) {
+	ks.AlertLock.Lock()
+	defer ks.AlertLock.Unlock()
+
+	delete(ks.AlertChans, id)
+
+	kg.Printf("Deleted client (%s) for ListenAlert", id)
+}
+
+func (ks *KoachServer) ListenAlert(req *proto.ListenAlertRequest, stream proto.ObservabilityService_ListenAlertServer) error {
+	id := uuid.Must(uuid.NewRandom()).String()
+
+	alertChan := make(chan *model.Alert, 8192)
+	defer close(alertChan)
+
+	ks.addAlertChans(id, alertChan)
+	defer ks.removeAlertChans(id)
+
+	for {
+		time.Sleep(time.Second * 1)
+
+		select {
+		case <-stream.Context().Done():
+			return nil
+		case msg, valid := <-alertChan:
+			if !valid {
+				continue
+			}
+
+			alertResponse := proto.ListenAlertResponse{
+				Message:  msg.Message,
+				Severity: int32(msg.Severity),
+				Observability: &proto.ObservabilityData{
```

Comment:
What's the need for separate DataType and manually creating it here?

Reply:
sure make sense, no need to add new data type

Reply:
wait, but isnt it better we seperate the data type for response and data type that we actually use in the program? in case we need some extension but not on the response